components/webext-storage/src/store.rs

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::api::{self, StorageChanges};
+use crate::db::StorageDb;
+use crate::error::*;
+use serde_json;
+use std::path::Path;
+use std::sync::Arc;
+
+use serde_json::Value as JsonValue;
+
+pub struct Store {
+    db: Arc<StorageDb>,
```

Comment:
I'm not sure this needs to be an `Arc`—only we (the `Store`) should have an owning reference to the connection. If we inline `StorageDb`'s methods into `Store`, this can become `Mutex<rusqlite::Connection>`, no reference counting needed.